native/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxTreeGenerator.java

```diff
@@ -466,6 +702,52 @@ private static Class createClientClass(Entity entity, String className, List<Nod
         delete.addStatement(NodeParser.parseStatement(BalFileConstants.DELETE_RUN_DELETE_QUERY));
         client.addMember(delete.getFunctionDefinitionNode(), true);
 
+        if (!entity.getRelations().isEmpty()) {
+            Function exists = new Function(BalFileConstants.EXISTS);
+            if (keys.size() > 1) {
+                keyString = new StringBuilder();
+                keyString.append("{");
+
+                for (String key : keys.keySet()) {
+                    keyString.append(entity.getEntityName().substring(0, 1).toLowerCase());
+                    keyString.append(entity.getEntityName().substring(1));
+                    keyString.append(".");
+                    keyString.append(key);
+                    keyString.append(", ");
+                }
+                keyString.append("}");
+
+            } else {
+                keyString = new StringBuilder();
+                keyString.append(entity.getEntityName().substring(0, 1).toLowerCase());
+                keyString.append(entity.getEntityName().substring(1));
+                keyString.append(".");
+                for (String key : keys.keySet()) {
+                    keyString.append(key);
+                }
+            }
+
+            exists.addRequiredParameter(TypeDescriptor.getQualifiedNameReferenceNode(BalFileConstants.ENTITIES,
+                            entity.getEntityName()), entity.getEntityName().substring(0, 1).toLowerCase() +
+                    entity.getEntityName().substring(1));
+            exists.addQualifiers(new String[]{BalFileConstants.KEYWORD_REMOTE});
+            exists.addReturns(TypeDescriptor.getUnionTypeDescriptorNode(
+                    TypeDescriptor.getBuiltinSimpleNameReferenceNode("boolean"),
+                    TypeDescriptor.getSimpleNameReferenceNode (BalFileConstants.ERROR)));
+            exists.addStatement(NodeParser.parseStatement(String.format(EXIST_READ_BY_KEY, entity.getEntityName(),
+                    keyString)));
+            IfElse typeCheck = new IfElse(NodeParser.parseExpression(String.format(CHECK_RESULT,
+                    entity.getEntityName())));
+            typeCheck.addIfStatement(NodeParser.parseStatement(RETURN_TRUE));
+            IfElse invalidKeyCheck = new IfElse(NodeParser.parseExpression(EXIST_CHECK_INVALID));
+            invalidKeyCheck.addIfStatement(NodeParser.parseStatement(RETURN_FALSE));
+            invalidKeyCheck.addElseStatement(NodeParser.parseStatement(String.format(RETURN_RESULT)));
```

Comment:
```suggestion
            invalidKeyCheck.addElseStatement(NodeParser.parseStatement(RETURN_RESULT));
```

Reply:
updated in [2bf9a5f](https://github.com/ballerina-platform/persist-tools/pull/23/commits/2bf9a5f4d1f39a2145630441c20b06f8d3c16ef9)